AccountManagementService.Domain/CommandHandlers/CreateTransactionCommandHandler.cs

```diff
@@ -0,0 +1,99 @@
+using AccountManagementService.Domain.Commands;
+using AccountManagementService.Domain.Contracts;
+using AccountManagementService.Domain.Models;
+using AccountManagementService.Domain.Notifications;
+using FluentValidation;
+using MediatR;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+
+namespace AccountManagementService.Domain.CommandHandlers;
+
+public class CreateTransactionCommandHandler : IRequestHandler<CreateTransactionCommand, object>
+{
+    private readonly IMediator _mediator;
+    private readonly ITransactionRepository _repository;
+    private readonly IAccountRepository _accountRepository;
+    private readonly IValidator<CreateTransactionCommand> _validator;
+
+    public CreateTransactionCommandHandler(IMediator mediator, ITransactionRepository repository, IAccountRepository accountRepository, IValidator<CreateTransactionCommand> validator)
+    {
+        _mediator = mediator;
+        _repository = repository;
+        _accountRepository = accountRepository;
+        _validator = validator;
+    }
+    
+    public async Task<object> Handle(CreateTransactionCommand request, CancellationToken cancellationToken)
+    {
+
+        try
+        {
+            await _validator.ValidateAndThrowAsync(request, cancellationToken);
+            
+            var transaction = new Transaction(request.TransactionType, request.Amount, request.CreditAccountId, request.DebitAccountId);
+            
+            await _repository.AddTransaction(transaction);
+            await _repository.Save();
+
+            if (request.CreditAccountId != null)
```

Comment:
Vi que tu usou o `is` em algumas partes do código, também conseguimos usar ele seguido de um `not` pra representar negação. Um exemplo (que talvez não se aplique a esse método), se o escopo do método usar mais `is` do que `==`, então é bom usar o `is not` ao invés de `!=` pra manter o mesmo nível semântico de abstração.
```suggestion
            if (request.CreditAccountId is not null)
```